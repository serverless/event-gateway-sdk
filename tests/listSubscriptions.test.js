const SDK = require('../lib/index')
const eventGatewayProcess = require('./utils/eventGatewayProcess')

const eventType = {
  space: 'testspace',
  name: 'pageVisited'
}

const functionConfig = {
  space: 'testspace',
  functionId: 'hello',
  type: 'awslambda',
  provider: {
    arn: 'arn::::',
    region: 'us-east-1'
  }
}

const subscriptionConfig = {
  space: 'testspace',
  subscriptionId: 'YXN5bmMscGFnZVZpc2l0ZWQsaGVsbG8sJTJGLFBPU1Q',
  type: 'async',
  functionId: 'hello',
  eventType: 'pageVisited',
  method: 'POST',
  path: '/'
}

let eventGateway
let eventGatewayProcessId

beforeAll(() =>
  eventGatewayProcess
    .spawn({
      configPort: 4001,
      apiPort: 4002
      // embedPeerPort: 4003,
      // embedCliPort: 4004,
    })
    .then(processInfo => {
      eventGatewayProcessId = processInfo.id
      eventGateway = new SDK({
        space: 'testspace',
        url: `http://localhost:${processInfo.apiPort}`,
        configurationUrl: `http://localhost:${processInfo.configPort}`
      })
    }))

afterAll(() => {
  eventGatewayProcess.shutDown(eventGatewayProcessId)
})

test('should create event type', () => {
  return eventGateway.createEventType(eventType).then(response => {
    expect(response).toEqual(eventType)
  })
})

test('should add a function to the gateway', () => {
  expect.assertions(1)
  return eventGateway.registerFunction(functionConfig).then(response => {
    expect(response).toEqual(functionConfig)
  })
})

test('should add a subscription to the gateway', () => {
  expect.assertions(1)
  return eventGateway.subscribe(subscriptionConfig).then(response => {
    expect(response).toMatchSnapshot()
  })
})

test('should return list of subscriptions', () => {
  expect.assertions(1)
  return eventGateway.listSubscriptions().then(response => {
    expect(response).toEqual([subscriptionConfig])
  })
})
