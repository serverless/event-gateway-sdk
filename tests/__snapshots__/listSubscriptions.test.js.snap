// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a subscription to the gateway 1`] = `
Object {
  "eventType": "pageVisited",
  "functionId": "hello",
  "method": "POST",
  "path": "/",
  "space": "testspace",
  "subscriptionId": "YXN5bmMscGFnZVZpc2l0ZWQsaGVsbG8sJTJGLFBPU1Q",
  "type": "async",
}
`;
